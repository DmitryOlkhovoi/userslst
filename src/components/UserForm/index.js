import React, { Component } from 'react'
import { Form, FormGroup, Label, Input, FormText, Button } from 'reactstrap'
import './styles/index.css'

class UserForm extends Component {
  state = {
    name: '',
    email: '',
    text: ''
  }

  constructor(props) {
    super(props)
    if (props.data) {
      this.state = props.data
    }
  }

  onSubmit = (e) => {
    e.preventDefault()
    this.props.onSubmit(this.state)
  }

  render() {
    return (
      <Form onSubmit={this.onSubmit}>
        <FormGroup>
          <Label for="exampleName">Name</Label>
          <Input
            name="name"
            id="exampleName"
            placeholder="name"
            defaultValue={this.state.name}
            onChange={(e) => this.setState({name: e.target.value})}
            required
          />
        </FormGroup>
        <FormGroup>
          <Label for="exampleEmail">Email</Label>
          <Input
            type="email"
            name="email"
            id="exampleEmail"
            placeholder="with a placeholder"
            defaultValue={this.state.email}
            onChange={(e) => this.setState({email: e.target.value})}
            required
          />
        </FormGroup>
        <FormGroup>
          <Label for="exampleText">Text Area</Label>
          <Input
            type="textarea"
            name="text"
            id="exampleText"
            defaultValue={this.state.text}
            onChange={(e) => this.setState({text: e.target.value})}
          />
        </FormGroup>
        <Button>Submit</Button>
      </Form>
    )
  }
}

export default UserForm
