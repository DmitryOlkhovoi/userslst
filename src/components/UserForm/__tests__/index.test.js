import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import UserForm from '../index'

it('renders correctly', () => {
  const tree = renderer.create(<UserForm />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with data', () => {
  const tree = renderer.create(<UserForm data={{
    name: 'foo',
    email: 'bar',
    text: 'lorem'
  }} />).toJSON()
  expect(tree).toMatchSnapshot()
})