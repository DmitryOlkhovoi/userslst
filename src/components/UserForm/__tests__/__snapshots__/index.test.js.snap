// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="exampleName"
    >
      Name
    </label>
    <input
      className="form-control"
      defaultValue=""
      id="exampleName"
      name="name"
      onChange={[Function]}
      placeholder="name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="exampleEmail"
    >
      Email
    </label>
    <input
      className="form-control"
      defaultValue=""
      id="exampleEmail"
      name="email"
      onChange={[Function]}
      placeholder="with a placeholder"
      required={true}
      type="email"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="exampleText"
    >
      Text Area
    </label>
    <textarea
      className="form-control"
      defaultValue=""
      id="exampleText"
      name="text"
      onChange={[Function]}
      type="textarea"
    />
  </div>
  <button
    className="btn btn-secondary"
    onClick={[Function]}
    type={undefined}
  >
    Submit
  </button>
</form>
`;

exports[`renders correctly with data 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="exampleName"
    >
      Name
    </label>
    <input
      className="form-control"
      defaultValue="foo"
      id="exampleName"
      name="name"
      onChange={[Function]}
      placeholder="name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="exampleEmail"
    >
      Email
    </label>
    <input
      className="form-control"
      defaultValue="bar"
      id="exampleEmail"
      name="email"
      onChange={[Function]}
      placeholder="with a placeholder"
      required={true}
      type="email"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="exampleText"
    >
      Text Area
    </label>
    <textarea
      className="form-control"
      defaultValue="lorem"
      id="exampleText"
      name="text"
      onChange={[Function]}
      type="textarea"
    />
  </div>
  <button
    className="btn btn-secondary"
    onClick={[Function]}
    type={undefined}
  >
    Submit
  </button>
</form>
`;
